ipaserver/plugins/ldap2.py

```diff
@@ -286,12 +286,16 @@ def get_effective_rights(self, dn, attrs_list):
 
         assert isinstance(dn, DN)
 
-        principal = getattr(context, 'principal')
-        entry = self.find_entry_by_attr("krbprincipalname", principal,
-            "krbPrincipalAux", base_dn=self.api.env.basedn)
+        bind_dn = None
+        try:
+            bind_dn = self.conn.whoami_s()[4:]
+        finally:
+            if bind_dn is None:
+                return None
```

Comment:
Why do we need a `try:..finally` block here? I would rather do `bind_dn = self.conn.whoami_s()[4:]` and let the exception propagate to the callers if something bad happens. Or you may catch it and re-raise some public and more descriptive error.